finder-app/writer.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+# Writer script for assignment 1
+# Author: Tim Bailey
+
+NUM_PARAMS=2
+writefile=$1
+writestr=$2
+
+# Check for correct number of input parameters
+if [ $# -ne $NUM_PARAMS ]
+then
+	echo "Incorrect number of parameters. Should be ${PARAMS}, was $#."
+	echo "The first argument is a path to a file on the filesystem."
+	echo "The second argument is a text string to write to the file."
+	echo "Exiting..."
+	exit 1
+fi
+
+# Check whether directory exists, create if not
+dir=$(dirname "$writefile")
+if [ ! -d "$dir" ]
+then
+	mkdir -v -p "$dir"
```

Comment:
@tiba6275 you could simplify the logic here and always just `mkdir -p`.  It won't fail if already existing.